Make install failing

```
TASK [chubbs : Make Install] **************************************************************************************************************************************************************************************
fatal: [***]: FAILED! => {"changed": false, "cmd": "/usr/bin/make install", "msg": "/usr/bin/ld: cannot find -lexpat\nclang: error: linker command failed with exit code 1 (use -v to see invocation)\nscons: *** [plugins/input/gdal.input] Error 1\nmake: *** [install] Error 2", "rc": 2, "stderr": "/usr/bin/ld: cannot find -lexpat\nclang: error: linker command failed with exit code 1 (use -v to see invocation)\nscons: *** [plugins/input/gdal.input] Error 1\nmake: *** [install] Error 2\n", "stderr_lines": ["/usr/bin/ld: cannot find -lexpat", "clang: error: linker command failed with exit code 1 (use -v to see invocation)", "scons: *** [plugins/input/gdal.input] Error 1", "make: *** [install] Error 2"], "stdout": "python3 scons/scons.py -j1 --config=cache --implicit-cache --max-drift=1 install\nscons: Reading SConscript files ...\n\u001b[94mUsing previous successful configuration...\u001b[0m\n\u001b[94mRe-configure by running \"python scons/scons.py configure\".\u001b[0m\n\u001b[94m\nWelcome to Mapnik...\n\u001b[0m\nscons: done reading SConscript files.\nscons: Building targets ...\nclang++-3.8 -o plugins/input/gdal.input -pthread -shared   plugins/input/gdal/gdal_datasource.os plugins/input/gdal/gdal_featureset.os $( -Ldeps/agg -Lsrc -Lsrc/json -Lsrc/wkt -Lmason_packages/.link/lib -Lmason_packages/linux-x86_64/libgdal/2.2.3/lib -Lmason_packages/linux-x86_64/proj/4.9.3/lib -Lmason_packages/linux-x86_64/sqlite/3.21.0/lib -Lmason_packages/linux-x86_64/expat/2.2.4/lib -Lmason_packages/linux-x86_64/jpeg_turbo/1.5.2 -Lmason_packages/linux-x86_64/jpeg_turbo/1.5.2/lib -Lmason_packages/linux-x86_64/libtiff/4.0.8/lib -Lmason_packages/linux-x86_64/libpng/1.6.32 -Lmason_packages/linux-x86_64/libpng/1.6.32/lib -Lmason_packages/linux-x86_64/libpq/9.6.5/lib -Lmason_packages/linux-x86_64/zlib/system -Lmason_packages/linux-x86_64/zlib/system/lib $) -lmapnik -lgdal -lz -lpthread -lm -lrt -ldl mason_packages/linux-x86_64/libpq/9.6.5/lib/libpq.a -lsqlite3 -ltiff -ljpeg -lproj -lpng -lexpat -lpq -lboost_system -licuuc\nscons: building terminated because of errors.\nMakefile:14: recipe for target 'install' failed\n", "stdout_lines": ["python3 scons/scons.py -j1 --config=cache --implicit-cache --max-drift=1 install", "scons: Reading SConscript files ...", "\u001b[94mUsing previous successful configuration...\u001b[0m", "\u001b[94mRe-configure by running \"python scons/scons.py configure\".\u001b[0m", "\u001b[94m", "Welcome to Mapnik...", "\u001b[0m", "scons: done reading SConscript files.", "scons: Building targets ...", "clang++-3.8 -o plugins/input/gdal.input -pthread -shared   plugins/input/gdal/gdal_datasource.os plugins/input/gdal/gdal_featureset.os $( -Ldeps/agg -Lsrc -Lsrc/json -Lsrc/wkt -Lmason_packages/.link/lib -Lmason_packages/linux-x86_64/libgdal/2.2.3/lib -Lmason_packages/linux-x86_64/proj/4.9.3/lib -Lmason_packages/linux-x86_64/sqlite/3.21.0/lib -Lmason_packages/linux-x86_64/expat/2.2.4/lib -Lmason_packages/linux-x86_64/jpeg_turbo/1.5.2 -Lmason_packages/linux-x86_64/jpeg_turbo/1.5.2/lib -Lmason_packages/linux-x86_64/libtiff/4.0.8/lib -Lmason_packages/linux-x86_64/libpng/1.6.32 -Lmason_packages/linux-x86_64/libpng/1.6.32/lib -Lmason_packages/linux-x86_64/libpq/9.6.5/lib -Lmason_packages/linux-x86_64/zlib/system -Lmason_packages/linux-x86_64/zlib/system/lib $) -lmapnik -lgdal -lz -lpthread -lm -lrt -ldl mason_packages/linux-x86_64/libpq/9.6.5/lib/libpq.a -lsqlite3 -ltiff -ljpeg -lproj -lpng -lexpat -lpq -lboost_system -licuuc", "scons: building terminated because of errors.", "Makefile:14: recipe for target 'install' failed"]}
```

was able to replicate this by sshing into server and running same command
```
root@<server-host-name>:/home/chubbs/mapnik# make install
python3 scons/scons.py -j1 --config=cache --implicit-cache --max-drift=1 install
scons: Reading SConscript files ...
Using previous successful configuration...
Re-configure by running "python scons/scons.py configure".
Welcome to Mapnik...
scons: done reading SConscript files.
scons: Building targets ...
clang++-3.8 -o test/unit/projection/proj_transform.o -c -std=c++14 -DU_USING_ICU_NAMESPACE=0 -D_GLIBCXX_USE_CXX11_ABI=0 -fvisibility=hidden -fvisibility-inlines-hidden -Wall -pthread -ftemplate-depth-300 -Wsign-compare -Wno-unsequenced -Wtautological-compare -Wheader-hygiene -O3 -DMAPNIK_MEMORY_MAPPED_FILE -DMAPNIK_HAS_DLCFN -DBIGINT -DBOOST_REGEX_HAS_ICU -DHAVE_JPEG -DMAPNIK_USE_PROJ -DMAPNIK_PROJ_VERSION=70201 -DHAVE_PNG -DHAVE_WEBP -DHAVE_TIFF -DLINUX -DMAPNIK_THREADSAFE -DBOOST_SPIRIT_NO_PREDEFINED_TERMINALS=1 -DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1 -DBOOST_SPIRIT_USE_PHOENIX_V3=1 -DNDEBUG -DMAPNIK_MEMORY_MAPPED_FILE -DMAPNIK_HAS_DLCFN -DBIGINT -DBOOST_REGEX_HAS_ICU -DHAVE_JPEG -DMAPNIK_USE_PROJ -DMAPNIK_PROJ_VERSION=70201 -DHAVE_PNG -DHAVE_WEBP -DHAVE_TIFF -DLINUX -DMAPNIK_THREADSAFE -DBOOST_SPIRIT_NO_PREDEFINED_TERMINALS=1 -DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1 -DBOOST_SPIRIT_USE_PHOENIX_V3=1 -DNDEBUG -DHAVE_CAIRO -DGRID_RENDERER -DSVG_RENDERER -DHAVE_CAIRO -Itest -Imason_packages/.link/include/cairo -Imason_packages/.link/include/pixman-1 -Ideps -Ideps/mapbox/polylabel/include -Ideps/mapbox/protozero/include -Ideps/mapbox/geometry/include -Ideps/mapbox/variant/include -Ideps/agg/include -Iinclude -Imason_packages/.link/include -Imason_packages/.link/include/freetype2 -Imason_packages/.link/include/libpng16 -Imason_packages/linux-x86_64/libgdal/2.2.3/include test/unit/projection/proj_transform.cpp
test/unit/projection/proj_transform.cpp:104:9: error: chosen constructor is explicit in copy-initialization
        {-4.0278869, 57.8796955}, // Dórnach, Highland
        ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:612:19: note: constructor declared here
        constexpr tuple(_U1&& __a1, _U2&& __a2)
                  ^
test/unit/projection/proj_transform.cpp:105:9: error: chosen constructor is explicit in copy-initialization
        {-4.2488787, 55.8609825}, // Glaschú, Alba
        ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:612:19: note: constructor declared here
        constexpr tuple(_U1&& __a1, _U2&& __a2)
                  ^
test/unit/projection/proj_transform.cpp:106:9: error: chosen constructor is explicit in copy-initialization
        {-1.4823897, 51.8726941}, // Charlbury, England
        ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:612:19: note: constructor declared here
        constexpr tuple(_U1&& __a1, _U2&& __a2)
                  ^
test/unit/projection/proj_transform.cpp:107:9: error: chosen constructor is explicit in copy-initialization
        {-3.9732612, 51.7077400}  // Felindre, Cymru
        ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../../include/c++/5.4.0/tuple:612:19: note: constructor declared here
        constexpr tuple(_U1&& __a1, _U2&& __a2)
                  ^
4 errors generated.
scons: *** [test/unit/projection/proj_transform.o] Error 1
scons: building terminated because of errors.
Makefile:14: recipe for target 'install' failed
make: *** [install] Error 2
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -7,7 +7,7 @@ on:
 jobs:
   test:
     name: Test
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-18.04
     steps:
       - name: Checkout code
         uses: actions/checkout@v2

--- a/defaults/main.yml
+++ b/defaults/main.yml
@@ -16,6 +16,8 @@ chubbs_apt_dependencies:
   - curl
   - git
   - unzip
+  - gcc-6
+  - g++-6
 
 chubbs_version: "master"
 chubbs_git_url: "https://github.com/onaio/PGRestAPI.git"

--- a/tasks/main.yml
+++ b/tasks/main.yml
@@ -27,6 +27,18 @@
     nvm_npm_path: "{{ chubbs_nvm_npm_path }}"
     nvm_version: "{{ chubbs_nvm_version }}"
 
+- name: Install software-properties-common
+  apt:
+    update_cache: yes
+    name: software-properties-common
+  tags:
+    - mapnik
+
+- name: Add ppa:ubuntu-toolchain-r/test into sources list
+  # noqa 305 301
+  shell: add-apt-repository -y ppa:ubuntu-toolchain-r/test
+  become: true
+
 - name: Install dependencies
   apt:
     update_cache: yes
@@ -42,6 +54,8 @@
     version: "{{ chubbs_mapnik_version }}"
     depth: 10
     dest: "{{ chubbs_mapnik_install_dir }}"
+    recursive: yes
+    update: yes
     force: yes
   tags:
     - mapnik

--- a/tasks/pm2.yml
+++ b/tasks/pm2.yml
@@ -56,6 +56,12 @@
   notify:
     - "reload pm2"
 
+- name: Cleanup pm2 processes
+  # noqa 305 301
+  become: true
+  become_user: "root"
+  shell: "killall -u {{ chubbs_system_user }}"
+
 - name: generate an PM2 init service file
   # noqa 305 301
   shell: "{{ chubbs_nvm_pm2_path }} startup {{ chubbs_init_system }} -u {{ chubbs_system_user }} --hp {{ chubbs_system_user_home }}"

